src/main/java/stsjorbsmod/patches/DimensionDoorPatch.java

```diff
@@ -0,0 +1,55 @@
+package stsjorbsmod.patches;
+
+import com.evacipated.cardcrawl.modthespire.lib.SpirePatch;
+import com.evacipated.cardcrawl.modthespire.lib.SpirePostfixPatch;
+import com.megacrit.cardcrawl.map.MapEdge;
+import com.megacrit.cardcrawl.map.MapRoomNode;
+import com.megacrit.cardcrawl.rooms.AbstractRoom;
+import com.megacrit.cardcrawl.scenes.AbstractScene;
+import java.util.Iterator;
+import stsjorbsmod.cards.DimensionDoor;
+
+public class DimensionDoorPatch {
+    @SpirePatch(
+            clz = MapRoomNode.class,
+            method = "isConnectedTo"
+    )
+    public static class MapRoomNode_isConnectedTo
+    {
+        @SpirePostfixPatch
+        public static boolean patch(boolean __result, MapRoomNode __this, MapRoomNode __node)
+        {
+            if (__result || !DimensionDoor.isMapTravelActive()) {
+                return __result;
+            }
+
+            // Any room on the next level is actually reachable.
+            Iterator edgeIterator = __this.getEdges().iterator();
```

Comment:
The `.iterator()` stuff is a common decompiler output, but I think it reads a littler clearer if you rewrite it as

```java
for (MapEdge edge : __this.getEdges()) {
    if (__node.y == edge.dstY) {
        return true;
    }
}
```

Reply:
In this case, the `stream()` based version is also pretty readable

```java
return __this.getEdges().stream().anyMatch(e -> __node.y == e.dstY);
```

Reply:
Thank you for connecting those dots! I had seen a use of stream in the Java docs but dismissed it as an IO thing. Hmm, I see now I went with a different formatting. It feels okay since functions weren't chained, but I understand if preferred to be in-line with your example style.